js_modules/dagit/packages/core/src/nav/ReloadRepositoryLocationButton.tsx

```diff
@@ -31,7 +38,7 @@ export const ReloadRepositoryLocationButton: React.FC<Props> = (props) => {
 
   return (
     <>
-      {children({tryReload, reloading})}
+      {children({tryReload, reloading, hasReloadPermission})}
```

Comment:
If you're expecting exactly 1 child with this interface then why not mark `Children` as it's own Component?

eg:
```
interface Props {
  ChildComponent: React.FC<ChildProps>;
  location: string;
}
```

then here you can do:
`<ChildComponent {...{tryReload, reloading, hasReloadPermission}}>`

This would allow the children to be a true component which would let them use React hooks

Reply:
I guess this pattern makes it a little bit easier to use the parent state so you don't need to pass that down.

Reply:
Actually I don't think I need any parent state at all for these. I'll give this a try.